## PR Review

Thanks for implementing stillbirth provisions for three more states in one tight PR. Formula structure, list-parameter integration, reference tuples, and the AZ-pattern parity are clean. A couple of issues to address.

### Critical (Must Fix)

**1. ND 2024 and 2025 inflation-adjusted values are missing**

`policyengine_us/parameters/gov/states/nd/tax/income/taxable_income/subtractions/stillborn.yaml` stops at `2023-01-01: 4_892`, so 2024+ simulations silently use the 2023 amount. The statute (N.D.C.C. § 57-38-30.3(2)) is inflation-adjusted annually and the ND Office of State Tax Commissioner has published:

- TY 2024 (Schedule ND-1SA Line 8): `$5,088`
- TY 2025 (Schedule ND-1SA Line 8): `$5,241`

The 2024 and 2025 booklets are already cited in `sources.yaml` so the references are in hand. Please extend:

```yaml
2024-01-01: 5_088
2025-01-01: 5_241
```

**2. CI status — all three `Full Suite` failures are runner cancellations, not code failures**

- `Full Suite - Baseline States` → cancelled mid-run at the `Run non-structural YAML tests (states in 3 batches)` step
- `Full Suite - Baseline (excl States) & Reform & Python` → cancelled at `Run non-structural YAML tests (other)`
- `Full Suite - Structural (Other)` → cancelled at `Run structural YAML tests (other)`

No test assertion failures appear in the logs; each job got a `##[error]The runner has received a shutdown signal` from Actions. Please push an empty commit (or re-run failed jobs) to get a clean CI signal before merge.

### Should Address

**3. Cite the codified NE statute, not just the bill**

`policyengine_us/parameters/gov/states/ne/tax/income/credits/stillborn.yaml` references LB 432 (2021) § 26 and the 1040N booklets. Add the codified statute as the primary reference:

```yaml
- title: Neb. Rev. Stat. § 77-2715.07(9)
  href: https://nebraskalegislature.gov/laws/statutes.php?statute=77-2715.07
```

Statute explicitly calls this a refundable credit, "For taxable years beginning or deemed to begin on or after January 1, 2022" — confirms the $2,000 value, the refundable classification, and the 2022 effective date.

**4. CT Public Act references need section numbers**

PA 22-118 is ~600 pages and PA 23-31 is sizable. The current references link to the full PDFs with no section or page anchor, so a reviewer has to scan the whole act. Please add the relevant section number (or `#page=` anchor) in each title so the reference points directly at the stillbirth-credit provision.

**5. PR description's "in Nebraska" claim is incorrect (description only, not code)**

The PR body says NE "specifically requires … that the stillbirth occurred IN NEBRASKA." Neb. Rev. Stat. § 77-2715.07(9) has no geographic requirement — that language is NE DOR booklet gloss on Neb. Rev. Stat. § 71-606(1) registration. The implementation is correct; just drop the "in Nebraska" requirement from the PR description to avoid future confusion.

**6. ND label and description should say "subtraction," not "deduction"**

Filename, class, folder, and the NDCC statute all use "subtraction." The variable `label = "North Dakota stillborn child deduction"` and the parameter's `label` / `description` use "deduction." For consistency and to match how the rest of the ND subtractions are named in this repo, change:

- `nd_stillborn_subtraction.py:7` → `label = "North Dakota stillborn child subtraction"`
- `parameters/.../stillborn.yaml` `label` and `description` → "subtraction"

**7. Cross-state-code isolation regression test**

Cheap insurance. Add one YAML case per state like:

```yaml
- name: Non-CT filer receives no CT stillborn credit
  period: 2023
  input:
    state_code: CA
    tax_unit_stillborn_children: 1
  output:
    ct_stillborn_credit: 0
```

`defined_for = StateCode.CT` enforces this already, but a case catches regressions when someone edits `defined_for` or the variable moves.

### Suggestions

**8. Document the eligibility simplification**

All three formulas reduce to `tax_unit_stillborn_children × per_stillbirth_amount`, pushing dependency / gestational-age / certificate-filing eligibility onto the input variable. This matches the AZ precedent, but consider noting in each variable's docstring that the input is assumed to be the count of *qualifying* stillbirths per the applicable state rules.

**9. Prefer HTML over PDF named-anchors for ND statute**

`https://ndlegis.gov/cencode/t57c38.pdf#nameddest=57-38-30p3` — PDF named-destinations don't always resolve in browsers. If ND offers an HTML-rendered chapter (or Justia's chapter view), consider adding that alongside the PDF.

**10. Integration / realistic-household YAML test**

Nothing in the test suite asserts that `ct_non_refundable_credits` / `ne_refundable_credits` / `nd_taxable_income` actually change when `tax_unit_stillborn_children > 0`. The list-parameter edits are correct (verified on branch), but a one-household end-to-end test would catch future wiring regressions.

---

### Validation Summary

| Check | Result |
|-------|--------|
| Regulatory Accuracy | 1 critical (ND 2024/2025 missing), 1 minor description issue |
| Reference Quality | 2 should (NE statute missing, CT PA section anchors) |
| Code Patterns | 1 should (ND subtraction vs. deduction naming) |
| Test Coverage | 2 should (cross-state isolation, realistic integration) |
| CI Status | 3 runner cancellations, not code failures — re-run required |

### Next Steps

- Add ND 2024 (`5_088`) and 2025 (`5_241`) values + references to the 2024/2025 ND-1SA Line 8 instructions.
- Add Neb. Rev. Stat. § 77-2715.07(9) to the NE reference tuple.
- Add PA section anchors to CT references.
- Rename "deduction" → "subtraction" in the ND variable label and parameter description.
- Add a cross-state-code isolation test per state.
- Re-run the three cancelled CI jobs.

Everything else — formula structure, AZ-pattern parity, list-parameter wiring, effective dates, parameter values for CT/NE and ND 2022/2023 — looks correct.

--- a/policyengine_us/parameters/gov/states/ct/tax/income/credits/stillborn.yaml
+++ b/policyengine_us/parameters/gov/states/ct/tax/income/credits/stillborn.yaml
@@ -13,3 +13,5 @@ metadata:
       href: https://www.cga.ct.gov/2022/act/pa/pdf/2022PA-00118-R00SB-00005-PA.pdf
     - title: Connecticut Public Act No. 23-31, § 10 (amended section 12-704i, effective June 7, 2023, applicable to taxable years commencing on or after January 1, 2022)
       href: https://www.cga.ct.gov/2023/act/pa/pdf/2023PA-00031-R00HB-06680-PA.pdf
+    - title: 2022 Schedule CT-IT Credit, Line 4 - Birth of a Stillborn Child Tax Credit
+      href: https://portal.ct.gov/-/media/drs/forms/2022/income/schedule-ct-it-credit_1222.pdf

--- a/policyengine_us/parameters/gov/states/ct/tax/income/credits/stillborn.yaml
+++ b/policyengine_us/parameters/gov/states/ct/tax/income/credits/stillborn.yaml
@@ -9,7 +9,7 @@ metadata:
   reference:
     - title: Conn. Gen. Stat. § 12-704i - Credit for delivery of a fetus born dead for which a fetal death certificate has been filed
       href: https://www.cga.ct.gov/current/pub/chap_229.htm#sec_12-704i
-    - title: Connecticut Public Act No. 22-118, § 411 (effective July 1, 2022, applicable to taxable years commencing on or after January 1, 2022)
+    - title: Connecticut Public Act No. 22-118, § 412 (enacted section 12-704i, effective July 1, 2022, applicable to taxable years commencing on or after January 1, 2022)
       href: https://www.cga.ct.gov/2022/act/pa/pdf/2022PA-00118-R00SB-00005-PA.pdf
-    - title: Connecticut Public Act No. 23-31 (amendment effective June 7, 2023, applicable to taxable years commencing on or after January 1, 2022)
+    - title: Connecticut Public Act No. 23-31, § 10 (amended section 12-704i, effective June 7, 2023, applicable to taxable years commencing on or after January 1, 2022)
       href: https://www.cga.ct.gov/2023/act/pa/pdf/2023PA-00031-R00HB-06680-PA.pdf

--- a/policyengine_us/parameters/gov/states/nd/tax/income/taxable_income/subtractions/sources.yaml
+++ b/policyengine_us/parameters/gov/states/nd/tax/income/taxable_income/subtractions/sources.yaml
@@ -7,6 +7,7 @@ values:
     - taxable_social_security
     - military_retirement_pay
     - nd_529_deduction
+    - nd_stillborn_subtraction
 
   2022-01-01:
     - us_govt_interest

--- a/policyengine_us/parameters/gov/states/nd/tax/income/taxable_income/subtractions/stillborn.yaml
+++ b/policyengine_us/parameters/gov/states/nd/tax/income/taxable_income/subtractions/stillborn.yaml
@@ -1,6 +1,6 @@
 description: North Dakota provides this subtraction from federal taxable income for each child who was stillborn during the tax year, provided a Fetal Death Record was filed with the North Dakota Division of Vital Records and the child would have been a dependent. The amount is adjusted annually for inflation.
 values:
-  2021-01-01: 0
+  2021-01-01: 4_310
   2022-01-01: 4_530
   2023-01-01: 4_892
   2024-01-01: 5_088
@@ -9,12 +9,15 @@ metadata:
   unit: currency-USD
   period: year
   label: North Dakota stillborn child subtraction amount
+  # ND indexes to Midwest CPI-U; IRS uprating is a proxy until available.
   uprating: gov.irs.uprating
   reference:
     - title: N.D.C.C. § 57-38-30.3(2) - Individual income tax - Subtraction for stillborn child
       href: https://ndlegis.gov/cencode/t57c38.pdf#nameddest=57-38-30p3
     - title: N.D.C.C. § 57-38-30.3(2) (Casetext rendering)
       href: https://casetext.com/statute/north-dakota-century-code/title-57-taxation/chapter-57-38-income-tax/section-57-38-303-individual-estate-and-trust-income-tax
+    - title: 2021 North Dakota Individual Income Tax Booklet, page 15, Line 8 instructions
+      href: https://www.tax.nd.gov/sites/www/files/documents/forms/individual/2021-iit/individual-income-tax-booklet-2021.pdf#page=15
     - title: 2022 Schedule ND-1SA Statutory Adjustments, Line 5 instructions
       href: https://www.tax.nd.gov/sites/www/files/documents/forms/individual/2022-iit/schedule-nd-1sa-2022.pdf#page=2
     - title: 2023 Schedule ND-1SA Statutory Adjustments, Line 8 instructions

--- a/policyengine_us/parameters/gov/states/nd/tax/income/taxable_income/subtractions/stillborn.yaml
+++ b/policyengine_us/parameters/gov/states/nd/tax/income/taxable_income/subtractions/stillborn.yaml
@@ -1,18 +1,27 @@
-description: North Dakota provides this deduction from federal taxable income for each child who was stillborn during the tax year, provided a Fetal Death Record was filed with the North Dakota Division of Vital Records and the child would have been a dependent. The amount is adjusted annually for inflation.
+description: North Dakota provides this subtraction from federal taxable income for each child who was stillborn during the tax year, provided a Fetal Death Record was filed with the North Dakota Division of Vital Records and the child would have been a dependent. The amount is adjusted annually for inflation.
 values:
   2021-01-01: 0
   2022-01-01: 4_530
   2023-01-01: 4_892
+  2024-01-01: 5_088
+  2025-01-01: 5_241
 metadata:
   unit: currency-USD
   period: year
-  label: North Dakota stillborn child deduction amount
+  label: North Dakota stillborn child subtraction amount
+  uprating: gov.irs.uprating
   reference:
-    - title: N.D.C.C. § 57-38-30.3(2) - Individual income tax - Deduction for stillborn child
+    - title: N.D.C.C. § 57-38-30.3(2) - Individual income tax - Subtraction for stillborn child
       href: https://ndlegis.gov/cencode/t57c38.pdf#nameddest=57-38-30p3
+    - title: N.D.C.C. § 57-38-30.3(2) (Casetext rendering)
+      href: https://casetext.com/statute/north-dakota-century-code/title-57-taxation/chapter-57-38-income-tax/section-57-38-303-individual-estate-and-trust-income-tax
     - title: 2022 Schedule ND-1SA Statutory Adjustments, Line 5 instructions
       href: https://www.tax.nd.gov/sites/www/files/documents/forms/individual/2022-iit/schedule-nd-1sa-2022.pdf#page=2
     - title: 2023 Schedule ND-1SA Statutory Adjustments, Line 8 instructions
       href: https://www.tax.nd.gov/sites/www/files/documents/forms/individual/2023-iit/schedule-nd-1sa-2023.pdf#page=2
+    - title: 2024 Schedule ND-1SA Statutory Adjustments, Line 8 instructions
+      href: https://www.tax.nd.gov/sites/www/files/documents/forms/individual/2024-iit/28710-schedule-nd-1sa-2024.pdf#page=2
+    - title: 2025 Schedule ND-1SA Statutory Adjustments, Line 8 instructions
+      href: https://www.tax.nd.gov/sites/www/files/documents/forms/individual/2025-iit/28710-schedule-nd-1sa-2025.pdf#page=2
     - title: 2022 North Dakota Individual Income Tax Booklet, page 17 "REMINDER - Deduction available to parents of stillborn child"
       href: https://www.tax.nd.gov/sites/www/files/documents/forms/individual/2022-iit/2022-individual-income-tax-booklet.pdf#page=17

--- a/policyengine_us/parameters/gov/states/ne/tax/income/credits/stillborn.yaml
+++ b/policyengine_us/parameters/gov/states/ne/tax/income/credits/stillborn.yaml
@@ -9,7 +9,7 @@ metadata:
   reference:
     - title: Neb. Rev. Stat. § 77-2715.07(9) (stillborn child tax credit)
       href: https://nebraskalegislature.gov/laws/statutes.php?statute=77-2715.07
-    - title: Nebraska LB 432 (2021), § 26 (enacting legislation)
+    - title: Nebraska LB 432 (2021), § 11 (enacting legislation)
       href: https://nebraskalegislature.gov/FloorDocs/107/PDF/Slip/LB432.pdf
     - title: 2022 Nebraska Individual Income Tax Booklet (Form 1040N, line 38 instructions)
       href: https://revenue.nebraska.gov/sites/revenue.nebraska.gov/files/doc/2022_Ne_Individual_Income_Tax_Booklet_8-307-2022_final_8.pdf#page=12

--- a/policyengine_us/parameters/gov/states/ne/tax/income/credits/stillborn.yaml
+++ b/policyengine_us/parameters/gov/states/ne/tax/income/credits/stillborn.yaml
@@ -7,7 +7,9 @@ metadata:
   period: year
   label: Nebraska stillborn child tax credit amount
   reference:
-    - title: Nebraska LB 432 (2021), § 26 (stillborn child tax credit, codified at Neb. Rev. Stat. § 77-2715.07)
+    - title: Neb. Rev. Stat. § 77-2715.07(9) (stillborn child tax credit)
+      href: https://nebraskalegislature.gov/laws/statutes.php?statute=77-2715.07
+    - title: Nebraska LB 432 (2021), § 26 (enacting legislation)
       href: https://nebraskalegislature.gov/FloorDocs/107/PDF/Slip/LB432.pdf
     - title: 2022 Nebraska Individual Income Tax Booklet (Form 1040N, line 38 instructions)
       href: https://revenue.nebraska.gov/sites/revenue.nebraska.gov/files/doc/2022_Ne_Individual_Income_Tax_Booklet_8-307-2022_final_8.pdf#page=12

--- a/policyengine_us/tests/policy/baseline/gov/states/ct/tax/income/credits/ct_stillborn_credit.yaml
+++ b/policyengine_us/tests/policy/baseline/gov/states/ct/tax/income/credits/ct_stillborn_credit.yaml
@@ -37,3 +37,11 @@
     tax_unit_stillborn_children: 1
   output:
     ct_stillborn_credit: 2_500
+
+- name: Non-CT filer with stillborn child receives no CT credit
+  period: 2023
+  input:
+    state_code: CA
+    tax_unit_stillborn_children: 1
+  output:
+    ct_stillborn_credit: 0

--- a/policyengine_us/tests/policy/baseline/gov/states/nd/tax/income/subtractions/nd_stillborn_subtraction.yaml
+++ b/policyengine_us/tests/policy/baseline/gov/states/nd/tax/income/subtractions/nd_stillborn_subtraction.yaml
@@ -1,4 +1,4 @@
-- name: No stillborn children, no deduction
+- name: No stillborn children, no subtraction
   period: 2023
   input:
     state_code: ND
@@ -6,7 +6,7 @@
   output:
     nd_stillborn_subtraction: 0
 
-- name: One stillborn child in 2022, $4,530 deduction
+- name: One stillborn child in 2022, $4,530 subtraction
   period: 2022
   input:
     state_code: ND
@@ -14,7 +14,7 @@
   output:
     nd_stillborn_subtraction: 4_530
 
-- name: Two stillborn children in 2022, $9,060 deduction
+- name: Two stillborn children in 2022, $9,060 subtraction
   period: 2022
   input:
     state_code: ND
@@ -22,7 +22,7 @@
   output:
     nd_stillborn_subtraction: 9_060
 
-- name: One stillborn child in 2023, $4,892 deduction
+- name: One stillborn child in 2023, $4,892 subtraction
   period: 2023
   input:
     state_code: ND
@@ -30,10 +30,34 @@
   output:
     nd_stillborn_subtraction: 4_892
 
-- name: Pre-effective year 2021, deduction is zero
+- name: One stillborn child in 2024, $5,088 subtraction
+  period: 2024
+  input:
+    state_code: ND
+    tax_unit_stillborn_children: 1
+  output:
+    nd_stillborn_subtraction: 5_088
+
+- name: One stillborn child in 2025, $5,241 subtraction
+  period: 2025
+  input:
+    state_code: ND
+    tax_unit_stillborn_children: 1
+  output:
+    nd_stillborn_subtraction: 5_241
+
+- name: Pre-effective year 2021, subtraction is zero
   period: 2021
   input:
     state_code: ND
+    tax_unit_stillborn_children: 1
+  output:
+    nd_stillborn_subtraction: 0
+
+- name: Non-ND filer with stillborn child receives no ND subtraction
+  period: 2023
+  input:
+    state_code: CA
     tax_unit_stillborn_children: 1
   output:
     nd_stillborn_subtraction: 0

--- a/policyengine_us/tests/policy/baseline/gov/states/nd/tax/income/subtractions/nd_stillborn_subtraction.yaml
+++ b/policyengine_us/tests/policy/baseline/gov/states/nd/tax/income/subtractions/nd_stillborn_subtraction.yaml
@@ -46,13 +46,13 @@
   output:
     nd_stillborn_subtraction: 5_241
 
-- name: Pre-effective year 2021, subtraction is zero
+- name: One stillborn child in 2021, $4,310 subtraction
   period: 2021
   input:
     state_code: ND
     tax_unit_stillborn_children: 1
   output:
-    nd_stillborn_subtraction: 0
+    nd_stillborn_subtraction: 4_310
 
 - name: Non-ND filer with stillborn child receives no ND subtraction
   period: 2023

--- a/policyengine_us/tests/policy/baseline/gov/states/ne/tax/income/credits/ne_stillborn_credit.yaml
+++ b/policyengine_us/tests/policy/baseline/gov/states/ne/tax/income/credits/ne_stillborn_credit.yaml
@@ -37,3 +37,11 @@
     tax_unit_stillborn_children: 1
   output:
     ne_stillborn_credit: 2_000
+
+- name: Non-NE filer with stillborn child receives no NE credit
+  period: 2023
+  input:
+    state_code: CA
+    tax_unit_stillborn_children: 1
+  output:
+    ne_stillborn_credit: 0

--- a/policyengine_us/variables/gov/states/nd/tax/income/subtractions/nd_stillborn_subtraction.py
+++ b/policyengine_us/variables/gov/states/nd/tax/income/subtractions/nd_stillborn_subtraction.py
@@ -4,7 +4,7 @@ from policyengine_us.model_api import *
 class nd_stillborn_subtraction(Variable):
     value_type = float
     entity = TaxUnit
-    label = "North Dakota stillborn child deduction"
+    label = "North Dakota stillborn child subtraction"
     unit = USD
     definition_period = YEAR
     defined_for = StateCode.ND

--- a/policyengine_us/variables/gov/states/ne/tax/income/credits/ne_stillborn_credit.py
+++ b/policyengine_us/variables/gov/states/ne/tax/income/credits/ne_stillborn_credit.py
@@ -8,7 +8,7 @@ class ne_stillborn_credit(Variable):
     unit = USD
     definition_period = YEAR
     defined_for = StateCode.NE
-    reference = "https://nebraskalegislature.gov/FloorDocs/107/PDF/Slip/LB432.pdf"
+    reference = "https://nebraskalegislature.gov/laws/statutes.php?statute=77-2715.07"
 
     def formula(tax_unit, period, parameters):
         p = parameters(period).gov.states.ne.tax.income.credits

